src/PriceBook.java

```diff
@@ -0,0 +1,6 @@
+import java.util.Hashtable;
+
+public class PriceBook {
+  // all attributes going through hash function to get HashString (String) that will lead to price (long)
+  private Hashtable<String, Long> book  = new Hashtable<>();
```

Comment:
let's store a product as a value

Reply:
Maybe hashString can already hash the Product and attributes so there Will be common Pricebook for all goods our Shop sells?

Reply:
You have to map that hash and proper product on that case